src/commands/timeout.ts

```diff
@@ -0,0 +1,160 @@
+            subcommand
+                .setName('remove')
+                .setDescription('Removes timeout from a user.')
+                .addUserOption((option) =>
+                    option.setName('user').setDescription('User to remove the timeout from').setRequired(true),
+                )
+                .addStringOption((option) =>
+                    option.setName('reason').setDescription('Reason for removing the timeout').setRequired(false),
+                ),
+        ),
+    run: async (interaction: ChatInputCommandInteraction) => {
+        // Typeguard in order to ensure having access to ChatInputCommand interaction options.
+        if (!interaction.isChatInputCommand()) return;
+
+        if (interaction.inGuild() === false) {
+            return interaction.reply({ content: "❌ I can't execute this command inside DMs!", ephemeral: true });
+        }
+
+        if (!interaction.guild.members.me.permissions.has(PermissionFlagsBits.ModerateMembers)) {
+            return interaction.reply({
+                content: '❌ Sorry, I need the `Moderate Members` in order to execute this command.',
+                ephemeral: true,
+            });
+        }
+        if (typeof interaction.member.permissions === 'string')
+            return interaction.reply({ content: '❌ Unknown permissions. Please try again later.', ephemeral: true });
+        else if (!interaction.member.permissions.has(PermissionFlagsBits.ModerateMembers)) {
+            return interaction.reply({
+                content: '⛔ You need the `Moderate Members` permission in order to use this command!',
+                ephemeral: true,
+            });
+        }
+
+        const user = interaction.options.getUser('user', true);
+        const member = await interaction.guild.members.fetch(user.id);
+
+        if (!member) {
+            return interaction.reply({
+                content: '❌ You need to specify a valid user to manage their timeout!',
+                ephemeral: true,
+            });
+        }
+
+        if (interaction.options.getSubcommand() === 'add') {
+            if (member.id === interaction.user.id) {
+                return interaction.reply({ content: "Aww, please don't time out yourself! 💖", ephemeral: true });
+            }
+
+            if (member.moderatable === false) {
+                return interaction.reply({
+                    content:
+                        '❌ I cannot time out this user! \n**Please make sure that my highest role is above theirs.**',
+                    ephemeral: true,
+                });
+            }
+
+            const duration = interaction.options.getString('duration', true);
+            const msDuration = ms(duration);
+
+            if (msDuration == null) {
+                return interaction.reply({
+                    content: '❌ The format of the timeout duration is invalid!',
+                    ephemeral: true,
+                });
+            }
```

Comment:
I feel like this wouldn't work out. Could you please show me what the bot sends when you try passing a duration that can't get parsed into a number on Discord?

Reply:
I have tested this, and the ms() method returns undefined when a string cannot be parsed. I used loosely equals null, because that checks for undefined (if the string cannot be parsed) and also for null (if the duration would be missing for some reason)

I can change it to === undefined, but I think this is the safest option

Reply:
Hmm. I'll have to experiment with this myself a little bit, if you don't mind. :)

Reply:
Let me know what you found! :)

Reply:
Just got back from testing the command, and while it does work, I was thinking that maybe the check could be done using `typeof duration !== 'number'` as the if condition? This would be consistent with most other checks in the bot's codebase. I'd send a suggestion as part of this comment, but sadly GitHub doesn't allow one to send suggestions on outdated comments.